src/test/coldstaking_tests.cpp

```diff
@@ -0,0 +1,46 @@
+// Copyright (c) 2020 The PIVX developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or https://www.opensource.org/licenses/mit-license.php.
+#include "test/test_pivx.h"
+
+#include "base58.h"
+#include "key.h"
+
+#include <boost/test/unit_test.hpp>
+
+BOOST_FIXTURE_TEST_SUITE(coldstaking_tests, BasicTestingSetup)
+
+// Goal: check cold staking script keys extraction
+BOOST_AUTO_TEST_CASE(extract_cold_staking_destination_keys)
+{
+    CKey ownerKey;
+    ownerKey.MakeNewKey(true);
+    CKey stakerKey;
+    stakerKey.MakeNewKey(true);
+    CScript script = GetScriptForStakeDelegation(ownerKey.GetPubKey().GetID(), stakerKey.GetPubKey().GetID());
+
+    // Check owner
+    CTxDestination ownerDest;
+    BOOST_CHECK(ExtractDestination(script, ownerDest, false));
+    BOOST_CHECK(boost::get<CKeyID>(&ownerDest));
```

Comment:
This only checks that the destination is valid (and we should use `IsValidDestination()` for that).
Would be good to compare it to the expected owner's key:
```cpp
    BOOST_CHECK(IsValidDestination(ownerDest));
    BOOST_CHECK(*boost::get<CKeyID>(&ownerDest) == ownerKey.GetPubKey().GetID());
```